src/util/wacz.ts

```diff
@@ -0,0 +1,424 @@
+      yield end;
+    }
+
+    async function* getData(data: Uint8Array) {
+      yield data;
+    }
+
+    for (const filename of files) {
+      const input = fs.createReadStream(filename);
+
+      const stat = await fsp.stat(filename);
+      const lastModified = stat.mtime;
+      const size = stat.size;
+
+      const nameStr = filename.slice(this.collDir.length + 1);
+      const name = encoder.encode(nameStr);
+
+      const currFile = new CurrZipFileMarker(filename, nameStr, size);
+
+      yield { input: wrapMarkers(currFile, input), lastModified, name, size };
+    }
+
+    // datapackage.json
+
+    const datapackageData = encoder.encode(
+      JSON.stringify(this.datapackage, null, 2),
+    );
+
+    yield {
+      input: getData(datapackageData),
+      lastModified: new Date(),
+      name: DATAPACKAGE_JSON,
+      size: datapackageData.length,
+    };
+
+    const hash =
+      "sha256:" + createHash("sha256").update(datapackageData).digest("hex");
+
+    // datapackage-digest.json
+
+    const digest: WACZDigest = {
+      path: DATAPACKAGE_JSON,
+      hash,
+    };
+
+    // Get Signature
+    if (this.signingUrl) {
+      const body = JSON.stringify({
+        hash,
+        created: this.datapackage.created,
+      });
+
+      const headers: Record<string, string> = {
+        "Content-Type": "application/json",
+      };
+
+      if (this.signingToken) {
+        headers["Authorization"] = this.signingToken;
+      }
+
+      try {
+        const response = await fetch(this.signingUrl, {
+          method: "POST",
+          headers,
+          body,
+        });
+        digest.signedData = await response.json();
+      } catch (e) {
+        logger.warn(
+          "Failed to sign WACZ, continuing w/o signature",
+          { ...formatErr(e) },
+          "wacz",
+        );
+      }
+    }
+
+    const digestData = encoder.encode(JSON.stringify(digest, null, 2));
+
+    yield {
+      input: getData(digestData),
+      lastModified: new Date(),
+      name: DATAPACKAGE_DIGEST_JSON,
+      size: digestData.length,
+    };
+  }
+}
+
+// Merge CDX
+export function addDirFiles(fullDir: string): string[] {
+  const files = fs.readdirSync(fullDir);
+  return files.map((name) => path.join(fullDir, name));
+}
+
+export async function mergeCDXJ(
+  warcCdxDir: string,
+  indexesDir: string,
+  zipped: boolean | null = null,
+) {
+  async function* readLinesFrom(stdout: Readable): AsyncGenerator<string> {
+    for await (const line of readline.createInterface({ input: stdout })) {
+      yield line + "\n";
+    }
+  }
+
+  async function* generateCompressed(
+    reader: AsyncGenerator<string>,
+    idxFile: Writable,
+  ) {
+    let offset = 0;
+
+    const encoder = new TextEncoder();
+
+    const filename = INDEX_CDX_GZ;
+
+    let cdxLines: string[] = [];
+
+    let key = "";
+    let count = 0;
+
+    idxFile.write(
+      `!meta 0 ${JSON.stringify({
+        format: "cdxj-gzip-1.0",
+        filename: INDEX_IDX,
```

Comment:
```suggestion
        filename: INDEX_CDX_GZ,
```

Small regression, this should point to the gzipped cdxj, not itself